src/CPUSolver.cpp

```diff
@@ -226,48 +145,29 @@ void CPUSolver::initializeFluxArrays() {
 void CPUSolver::initializeSourceArrays() {
 
   /* Delete old sources arrays if they exist */
-  if (_fission_sources != NULL)
-    delete [] _fission_sources;
-
-  if (_scatter_sources != NULL)
-    delete [] _scatter_sources;
-
-  if (_old_fission_sources != NULL)
-    delete [] _old_fission_sources;
-
   if (_reduced_sources != NULL)
     delete [] _reduced_sources;
 
-  if (_source_residuals != NULL)
-    delete [] _source_residuals;
-
-  int size;
-
   /* Allocate memory for all source arrays */
   try{
-
-    size = _num_FSRs * _num_groups;
-    _fission_sources = new FP_PRECISION[size];
+    int size = _num_FSRs * _num_groups;
     _reduced_sources = new FP_PRECISION[size];
 
-    size = _num_threads * _num_groups;
-    _scatter_sources = new FP_PRECISION[size];
-
-    size = _num_FSRs;
-    _old_fission_sources = new FP_PRECISION[size];
-    _source_residuals = new FP_PRECISION[size];
-
+    /* If no fixed sources were assigned, use a zeroes array */
+    if (_fixed_sources == NULL) {
+      _fixed_sources = new FP_PRECISION[size];
+      memset(_fixed_sources, 0.0, sizeof(FP_PRECISION) * size);
+    }
```

Comment:
This seems like a good solution for fixed source problems, but I don't like it happening for a large eigenvalue calculation.  Could we move this to the fixed source and subcritical solver routines rather than a fully shared routine?